src/convertToObject.js

```diff
@@ -6,7 +6,25 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const mainObject = {};
+  const trimmedString = sourceString.trim();
+
+  const arrayFromString = trimmedString.split(';');
+
+  arrayFromString.forEach((element) => {
+    if (element.includes(':')) {
+      const [property, value] = element.split(':');
+
+      if (property && value) {
```

Comment:
Consider checking for non-empty trimmed values for both `property` and `value` to avoid adding empty keys or values to the `mainObject`. This will ensure that only valid key-value pairs are added.